### PR TITLE
Generic client runtime config

### PR DESCRIPTION
**Pull request by srchase**
This PR updates client configuration to use the client config that was moved to smithy-typescript in https://github.com/awslabs/smithy-typescript/pull/693

Clients were re-generated, due to the change in ordering of the generated config, but the content is the same.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java [hunk 1]
```diff
@@ -74,15 +74,6 @@ public void addConfigInterfaceFields(
             SymbolProvider symbolProvider,
             TypeScriptWriter writer
     ) {
-        writer.addImport("Provider", "__Provider", TypeScriptDependency.AWS_SDK_TYPES.packageName);
```
Comment (gosar) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java:77]: Update the javadoc to reflect the changes.

File: codegen/smithy-aws-typescript-codegen/src/test/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfigTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public void awsClient() {
 
         // Check dependencies
         assertThat(manifest.getFileString("package.json").get(),
-                   containsString(AwsDependency.NODE_CONFIG_PROVIDER.packageName));
+                   containsString(TypeScriptDependency.NODE_CONFIG_PROVIDER.packageName));
```
Comment (gosar) [inline codegen/smithy-aws-typescript-codegen/src/test/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfigTest.java:38]: update the tests to not assert for things that are not part of its logic.

**Review by gosar — COMMENTED** [review]
See comments. I had paused before I got to those.

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java [hunk 3]
```diff
@@ -74,15 +74,6 @@ public void addConfigInterfaceFields(
...
-
-        writer.writeDocs("Value for how many times a request will be made at most in case of retry.")
-                .write("maxAttempts?: number | __Provider<number>;\n");
-        writer.writeDocs("Specifies which retry algorithm to use.")
-                .write("retryMode?: string | __Provider<string>;\n");
-        writer.writeDocs("Optional logger for logging debug/info/warn/error.")
-                .write("logger?: __Logger;\n");
         writer.writeDocs("Enables IPv6/IPv4 dualstack endpoint.")
                 .write("useDualstackEndpoint?: boolean | __Provider<boolean>;\n");
         writer.writeDocs("Enables FIPS compatible endpoints.")
```
Comment (gosar) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java:79]: While we are updating this, let's make the Integration be appropriately conditional. On the aws.api#service trait.

Comment (gosar) [reply codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java:79]: In fact, given these are AWS specific, maybe they should just be folded into AddAwsRuntimeConfig.

Comment (srchase) [reply codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddClientRuntimeConfig.java]: Moved into AddAwsRuntimeConfig and made appropriately conditional in [3e1cb31](https://github.com/aws/aws-sdk-js-v3/pull/4427/commits/3e1cb31c48c5b75a90cbd69f9e638070afecaaa1).

**Review by gosar — APPROVED** [review]

**Review by kuhe — APPROVED** [review]

**Pull request merged**